Source/missiles.cpp

```diff
@@ -3848,31 +3845,27 @@ void ProcessFlameWaveControl(Missile &missile)
 
 void ProcessRage(Missile &missile)
 {
-	missile.duration--;
-
-	if (missile.duration != 0) {
-		return;
-	}
+	if (--missile.duration == 0) {

```

Comment:
I think it was more readable before. Reversing the condition reduces nesting, and I'm generally pretty hesitant to combine increments with other statements. In this case, the conditional expression is pretty simple, but perhaps it's worth noting that it's a recipe for easy mistakes in general: `if (IsSomething() && --missile.duration == 0)`